distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso

```diff
@@ -310,68 +317,112 @@ type SQLServer_Dialect
+        pair typed_result [wrapped_name]
+    SQL_Expression.Constant value -> 
+        wrapped = case value of
+            Nothing -> SQL_Builder.code "NULL"
+            _ -> SQL_Builder.interpolation value
+        pair wrapped [wrapped]
+    SQL_Expression.Literal value -> 
+        wrapped = SQL_Builder.code value
+        pair wrapped [wrapped]
+    SQL_Expression.Text_Literal _ ->
+        wrapped_literal = base_gen.generate_expression dialect expr
+        pair wrapped_literal []
+    SQL_Expression.Operation kind arguments metadata ->
+        op = dialect.dialect_operations.operations_dict.get kind (Error.throw <| Unsupported_Database_Operation.Error kind)
+        requires_bool = _op_requires_bool kind
+        parsed_args_and_null_checks = arguments.map (c -> _generate_expression dialect base_gen c requires_bool)
+        parsed_args = parsed_args_and_null_checks.map .first
+
+        null_checks = parsed_args_and_null_checks.map .second  . flatten
+
+        expr_result = case kind of
+            ## In the case that we actually want to check for null then we need to generate the null
+                check sql for all the columns that have been used up to this point and that 
+                becomes the expression.
+            "IS_NULL" -> _generate_null_check_sql_builder null_checks
+            _ -> 
+                result = op parsed_args
+                # If the function expects more arguments, we pass the metadata as the last argument.
+                case result of
+                    _ : Function -> result metadata
+                    _ -> result
+        null_checks_result = if kind == "IS_NULL" then [] else null_checks
+        typed_expr = if _op_returns_boolean kind == generate_as_bool then expr_result else
+            if generate_as_bool then expr_result ++ " = 1" else
+                SQL_Builder.code "CASE WHEN " ++ expr_result ++ " THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT) END"
+
+        pair typed_expr null_checks_result
+    query : Query -> pair (base_gen.generate_sub_query dialect query) []
+
+private _generate_column dialect base_gen expr:(SQL_Expression | Order_Descriptor | Query) -> SQL_Builder =
+    case expr of
+        SQL_Expression.Operation kind _ _ ->
+            generating_a_boolean_column = _op_returns_boolean kind
+            expr_null_checks_pair = _generate_expression dialect base_gen expr generate_as_bool=generating_a_boolean_column
+            base_expr = expr_null_checks_pair.first
+            null_check = if expr_null_checks_pair.second.length == 0 then "" else
+                SQL_Builder.code "WHEN " ++ _generate_null_check_sql_builder expr_null_checks_pair.second ++ " THEN NULL "
+            case generating_a_boolean_column of
+                False -> SQL_Builder.code "CASE " ++ null_check ++ " ELSE " ++ base_expr ++ " END"
+                True -> SQL_Builder.code "CASE " ++ null_check ++ "WHEN " ++ base_expr ++ " THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT) END"
```

Comment:
I see multiple places with the `THEN CAST(1 AS BIT) ELSE ...` code. Could we refactor it to a single common 'bool condition to value' conversion? I think it will not only be 'DRY' but also make the code more readable as it will be explicit where the conversion happens.

Reply:
Done